pkg/formats/sniffer.go

```diff
@@ -103,27 +146,13 @@ func (c cdxSniff) sniff(data []byte) Format {
 		state.Encoding = JSON
 	}
 
-	if state.Encoding == JSON {
-		type SpecVersionStruct struct {
-			SpecVersion string `json:"specVersion"`
-		}
-
-		var specversionjson SpecVersionStruct
-		err := json.Unmarshal(data, &specversionjson)
-		if err == nil {
-			state.Version = specversionjson.SpecVersion
-		}
-	}
-
-	if state.Version == "" || state.Encoding == "" {
-		if strings.Contains(stringValue, `"specVersion"`) {
-			parts := strings.Split(stringValue, ":")
-			if len(parts) == 2 {
-				ver := strings.TrimPrefix(strings.TrimSuffix(strings.TrimSuffix(strings.TrimSpace(parts[1]), ","), "\""), "\"")
-				if ver != "" {
-					state.Version = ver
-					state.Encoding = JSON
-				}
+	if strings.Contains(stringValue, `"specVersion"`) {
+		parts := strings.Split(stringValue, ":")
+		if len(parts) == 2 {
+			ver := strings.TrimPrefix(strings.TrimSuffix(strings.TrimSuffix(strings.TrimSpace(parts[1]), ","), "\""), "\"")
+			if ver != "" {
+				state.Version = ver
+				state.Encoding = JSON
```

Comment:
Now that we've handled JSON parsing using the universal struct, I think we could remove all the code that handles JSON in the string parser.

We can do this in a follow up too.

Reply:
Yes, good idea.  I added a commit to remove the strings-based parsing from both sniff functions - the entire content of the cdxSniff version and the portion of spdxSniff.  Thanks!